apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/bytebuddy/ClassNameParser.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.sdk.bytebuddy;
+
+/**
+ * Utility class to extract the correct simple class name
+ */
+public class ClassNameParser {
+    /**
+     * Utility class, do not instantiate
+     */
+    private ClassNameParser() {}
+    /**
+     * Parses the simple class name from a class name
+     * @param className
+     * @return
+     */
+    public static String parse(String className) {
+        //remove package name
+        className = className.substring(className.lastIndexOf('.') + 1);
+        if(className.contains("$")) {
+            if(className.endsWith("$")) {
+                // this can happen if the source code was in Scala and the object keyword was used
+                // https://www.toptal.com/scala/scala-bytecode-and-the-jvm
+                className = className.substring(0, className.length() - 1);
```

Comment:
What about nested objects, does this exist in Scala? E.g. `Outer$Inner$`

Reply:
I don't really know as I am not familiar with scala. Maybe the issue creator knows this @georgms

Reply:
Okay, I tried it out with https://scastie.scala-lang.org/ :

Nested classes is the same as in Java:  
```scala
class Graph:
  class Node:
    var connectedNodes: List[Node] = Nil
    def connectTo(node: Node): Unit =
      if !connectedNodes.exists(node.equals) then
        connectedNodes = node :: connectedNodes

  var nodes: List[Node] = Nil
  def newNode: Node =
    val res = Node()
    nodes = res :: nodes
    res

var g = Graph()
var n = g.newNode

Console.println(n.getClass().getName())
```

Output (removed Playground prefix which was added in  the web debugger):
> Graph$Node

For nested Objects, the same rules apply:  
```scala
import java.io.ObjectOutput
object Graph:
  object Node:
    def helloWorld(): Unit =
      Console.println("Hello World")


Console.println(Graph.Node.getClass().getName())
```

Output (removed Playground prefix which was added in  the web debugger):
> Graph$Node$